File: AndroidStudioProjects/lab04/app/src/main/java/pbs/edu/lab04/SimpleListsActivity.java
package pbs.edu.lab04;

import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class SimpleListsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_simple_lists);

        ListView carList = findViewById(R.id.carList);
        final String[] cars = {"Toyota", "BMW", "Audi", "Mercedes", "Ford", "Honda", "Tesla", "Volkswagen", "Volvo", "Mazda",
                "Ferrari", "Lamborghini", "Porsche", "Bugatti", "Maserati", "Jaguar"};

        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, cars);
        carList.setAdapter(adapter);

        carList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Toast.makeText(getApplicationContext(),
                        "Wybrano element " + position + ", czyli " + cars[position],
                        Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_ESCAPE) {
            finish();
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: AndroidStudioProjects/lab05/app/src/main/java/pbs/edu/lab05/SimpleListsActivity.kt
package pbs.edu.lab05

import android.os.Bundle
import android.view.KeyEvent
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class SimpleListsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_simple_lists)

        val carList = findViewById<ListView>(R.id.carList)
        val cars = arrayOf(
            "Toyota", "BMW", "Audi", "Mercedes", "Ford", "Honda", "Tesla",
            "Volkswagen", "Volvo", "Mazda", "Ferrari", "Lamborghini", "Porsche",
            "Bugatti", "Maserati", "Jaguar"
        )

        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, cars)
        carList.adapter = adapter

        carList.onItemClickListener =
            AdapterView.OnItemClickListener { _, _, position, _ ->
                val selectedCar = cars[position]
                Toast.makeText(applicationContext, "Wybrano: $selectedCar", Toast.LENGTH_SHORT).show()
            }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        if (keyCode == KeyEvent.KEYCODE_ESCAPE) {
            finish()
            return true
        }
        return super.onKeyDown(keyCode, event)
    }
}
